bomi/datastructure.py

```diff
@@ -108,12 +116,35 @@ def add_packet(self, packet: Packet):
         self.sensor_fp.write(",".join((str(v) for v in (packet.time, *readings))) + "\n")
 
         # Shift buffer when full, never changing buffer size
-        self.data[:-1] = self.data[1:]
-        self.data[-1] = readings
+        self._raw_data[:-1] = self._raw_data[1:]
+        self._raw_data[-1] = readings
         self.timestamp[:-1] = self.timestamp[1:]
         self.timestamp[-1] = packet.time
 
 
+class AveragedMultichannelBuffer(MultichannelBuffer):
```

Comment:
I think we should add unit tests! This class is pretty close to [SRP](https://en.wikipedia.org/wiki/Single-responsibility_principle) and should be pretty easy to test. For code quality and maintainability, we can begin adding unit tests to SRP code snippets. Later, we should come up with some protocol for integration tests (with Trigno/Yost etc.) Mocking the Yost and Trigno API would be cool, but may be too much work to maintain the mocks themselves. This kind of testing is important to prevent what happened to us last time - you modify a piece of code, and it unintendedly caused a bug somewhere else in the program that we didn't immediate catch manually.

An easy starting point is to simply put unittest cases in the same file as the implementation, and later move to a separate tests directory if desired. For example, I've done this:

https://github.com/OpticalUltrasoundImaging/oct_labeler/blob/5e347df8f98fe54b016c0177d3fe65bc200cd69b/oct_labeler/data.py#L484-L521

After adding unit tests, we can compute test coverage automatically, and test manually if something outside the testing coverage is changed.

Reply:
You're right, I'll start working on unit tests.